src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,639 @@
+            }
+            switch (ig.lightingMethod) {
+            case LightingMethod::LED: {
+                ig.ZPPFD = ScheduleManager::GetCurrentScheduleValue(state, ig.SchedLEDPtr) * ig.LEDNominalPPFD; // PPFD
+                ig.LEDActualPPFD = ig.LEDNominalPPFD;
+                ig.LEDActualEleP = ig.LEDNominalEleP;
+                ig.LEDActualEleCon = ig.LEDNominalEleP * Timestep;
+            } break;
+            case LightingMethod::Daylighting: {
+                ig.ZPPFD = 0;
+                ig.LEDActualPPFD = 0;
+                ig.LEDActualEleP = 0;
+                ig.LEDActualEleCon = 0;
+                if (!state.dataDayltg->CalcDayltghCoefficients_firstTime && state.dataEnvrn->SunIsUp) {
+                    ig.ZPPFD = state.dataDayltg->daylightControl(ig.LightControlPtr).refPts(1).lums[DataSurfaces::iLum_Illum] /
+                               77; // To be updated currently only take one reference point; 77 conversion factor from Lux to PPFD
+                }
+            } break;
+            case LightingMethod::LEDDaylighting: {
+                Real64 a = ScheduleManager::GetCurrentScheduleValue(state, ig.SchedLEDDaylightTargetPtr);
+                Real64 b = 0;
+                if (!state.dataDayltg->CalcDayltghCoefficients_firstTime && state.dataEnvrn->SunIsUp) {
+                    b = state.dataDayltg->daylightControl(ig.LightControlPtr).refPts(1).lums[DataSurfaces::iLum_Illum] /
+                        77; // To be updated currently only take one reference point; 77 conversion factor from Lux to PPFD
+                }
+                ig.LEDActualPPFD = max((a - b), 0.0);
+                if (ig.LEDActualPPFD >= ig.LEDNominalPPFD) {
+                    ig.ZPPFD = ig.LEDNominalPPFD + b; // LED Nominal + Daylight
+                    ig.LEDActualEleP = ig.LEDNominalEleP;
+                    ig.LEDActualEleCon = ig.LEDNominalEleP * Timestep;
+                } else {
+                    ig.ZPPFD = a; // Targeted PPFD
+                    ig.LEDActualEleP = ig.LEDNominalEleP * ig.LEDActualPPFD / ig.LEDNominalPPFD;
+                    ig.LEDActualEleCon = ig.LEDActualEleP * Timestep;
+                }
+            } break;
+            default:
+                break;
+            }
+            ZonePPFD = ig.ZPPFD;
+            ZoneVPD = ig.ZVPD / 1000; // kPa
+            // ET Calculation
+            switch (ig.etCalculationMethod) {
+            case ETCalculationMethod::PenmanMonteith: {
+                Real64 SwitchF = 1.0;
+                ig.ETRate = ETBaseFunction(state, ZonePreTemp, ZonePreHum, ZoneCO2, ZonePPFD, ZoneVPD, LAI, SwitchF);
+            } break;
+            case ETCalculationMethod::Stanghellini: {
+                Real64 SwitchF = 2 * LAI;
+                ig.ETRate = ETBaseFunction(state, ZonePreTemp, ZonePreHum, ZoneCO2, ZonePPFD, ZoneVPD, LAI, SwitchF);
+            } break;
+            default:
+                break;
+            }
+            if (ig.EMSETCalOverrideOn) {
```

Comment:
This should be done _before_ the switch statement above, that way if the user is overriding the value, it won't waste calculation time.  Also...what is happening here if you don't use EMS...is it failing?

Reply:
Agreed that save time to move in front of switch statement. Run successfully with PenmanMonteith and Stanghellini methods.